app/code/Magento/ImportExport/Model/Import/Source/FileParser/ZipParserFactory.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+
+class ZipParserFactory implements ParserFactoryInterface
+{
+    private $filesystem;
+    private $isZipAvailable;
+    private $parserFactory;
+
+    public function __construct(
+        Filesystem $filesystem,
+        ParserFactoryInterface $parserFactory,
+        $isZipAvailable = null
+    ) {
+        $this->filesystem = $filesystem;
+        $this->parserFactory = $parserFactory;
+        $this->isZipAvailable = $isZipAvailable ?? extension_loaded('zip');
+    }
+
+    public function create($path, array $options = [])
+    {
+        $this->assertZipFileIsParsable($path);
+
+        $sourceDirectory = $this->filesystem->getDirectoryRead(DirectoryList::ROOT);
+        $writeDirectory = $this->filesystem->getDirectoryWrite(DirectoryList::TMP);
+
+        $path = $sourceDirectory->getRelativePath($path);
+        $this->assertFileExistance($path, $sourceDirectory);
+
+        $zip = new \ZipArchive();
+
+        if ($zip->open($sourceDirectory->getAbsolutePath($path)) !== true) {
+            throw new CorruptedFileException($path);
+        }
+
+        $files = $this->fetchFileNamesFromZipFile($zip);
+
+        foreach ($files as $file) {
```

Comment:
Because it is standard behavior for merchants to do. User experience must be taken into account, so no special file naming and crafting of zip archive should be used, it should be simple and intuitive. If you want to import products, you should be able just to compress any file or folder with a file and import it into Magento. That is why  code must be smart enough to ignore non valid files like OS specific ".Thumbs", or not supported files.